packages/tools/examples/viewReferencePresentation/index.ts

```diff
@@ -0,0 +1,517 @@
+displayAreaOptions.set('Scale Right Bottom', scaleRightBottom);
+
+let storeAsInitialCamera = true;
+addToggleButtonToToolbar({
+  id: 'storeAsInitialCameraToggle',
+  title: 'Store Display Area',
+  defaultToggle: storeAsInitialCamera,
+  onClick: (toggle) => {
+    storeAsInitialCamera = toggle;
+  },
+});
+
+addDropdownToToolbar({
+  id: 'displayArea',
+  options: {
+    values: Array.from(displayAreaOptions.keys()),
+    defaultValue: displayAreaOptions.keys().next().value,
+  },
+  onSelectedValueChange: (value) => {
+    const displayArea = displayAreaOptions.get(value);
+    displayArea.storeAsInitialCamera = storeAsInitialCamera;
+    viewport.setDisplayArea(displayArea);
+    viewport.render();
+  },
+});
+
+addButtonToToolbar({
+  title: 'Reset Camera',
+  onClick: () => {
+    viewport.resetCamera();
+  },
+});
+
+const rotations = ['0', '45', '90', '135', '180', '270'];
+addDropdownToToolbar({
+  id: 'rotation',
+  options: {
+    values: rotations,
+    defaultValue: rotations[0],
+  },
+  onSelectedValueChange: (value) => {
+    viewport.setProperties({ rotation: value });
+    viewport.render();
+  },
+});
+
+// ============================= //
+let withOrientation = false;
+
+addToggleButtonToToolbar({
+  id: 'applyOrientation',
+  title: 'Apply Orientation',
+  defaultToggle: withOrientation,
+  onClick: (toggle) => {
+    withOrientation = toggle;
+  },
+});
+
+let applyPresentation = false;
+
+addToggleButtonToToolbar({
+  id: 'applyPresentation',
+  title: 'Apply Presentation',
+  defaultToggle: applyPresentation,
+  onClick: (toggle) => {
+    applyPresentation = toggle;
+  },
+});
+
+let resizeTimeout = null;
+const resizeObserver = new ResizeObserver(() => {
+  if (resizeTimeout) {
+    return;
+  }
+  resizeTimeout = setTimeout(resize, 100);
+});
+
+function resize() {
+  resizeTimeout = null;
+  const renderingEngine = getRenderingEngine(renderingEngineId);
+
+  if (renderingEngine) {
+    const presentations = viewports.map((viewport) =>
+      viewport.getViewPresentation()
+    );
+    renderingEngine.resize(true, false);
+    viewports.forEach((viewport, idx) => {
+      viewport.setView(null, presentations[idx]);
+    });
+  }
+}
+
+/**
+ * Handles the sync between the source and destination viewport
+ */
+function viewportRenderedListener(event) {
+  const { element } = event.detail;
+  const { viewport } = getEnabledElement(element);
+  const viewRef = viewport.getViewReference();
+  const viewPres = viewport.getViewPresentation();
+  for (const destElement of setViewElements) {
+    const { viewport: destViewport } = getEnabledElement(destElement);
+    if (
+      destViewport.isReferenceViewable(viewRef, {
+        withNavigation: true,
+        asVolume: false,
+        withOrientation,
+      })
+    ) {
+      destViewport.setView(viewRef, viewPres);
+      destViewport.render();
+    } else if (applyPresentation) {
+      destViewport.setView(null, viewPres);
+      destViewport.render();
+    }
+  }
+}
```

Comment:
I thought you had a synchronizer. I'm surprised to see this issue here; perhaps we need a `PresentationSynchronizer`.

Reply:
Added notes explaining what this is doing.